packages/fuel-indexer-schema/src/db/queries.rs

```diff
@@ -0,0 +1,305 @@
+                            })
+                        })
+                        .collect::<Vec<String>>()
+                        .join(" AND ");
+                    query = format!("{query} WHERE {filter_str}");
+                }
+
+                println!("{query}");
+                query
+            }
+        }
+    }
+
+    fn parse_query_elements(&self, db_type: &DbType) -> Vec<String> {
+        let mut peekable_elements = self.elements.iter().peekable();
+
+        let mut elements = Vec::new();
+
+        match db_type {
+            DbType::Postgres => {
+                while let Some(e) = peekable_elements.next() {
+                    match e {
+                        // Set the key for this JSON element to the name of the entity field
+                        // and the value to the corresponding database table so that it can
+                        // be successfully retrieved.
+                        QueryElement::Field { key, value } => {
+                            elements.push(format!("'{key}', {value}"));
+
+                            // If the next element is not a closing boundary, then a comma should
+                            // be added so that the resultant SQL query can be properly constructed.
+                            if let Some(next_element) = peekable_elements.peek() {
+                                match next_element {
+                                    QueryElement::Field { .. }
+                                    | QueryElement::ObjectOpeningBoundary { .. } => {
+                                        elements.push(", ".to_string());
+                                    }
+                                    _ => {}
+                                }
+                            }
+                        }
+
+                        // Set a nested JSON object as the value for this entity field.
+                        QueryElement::ObjectOpeningBoundary { key } => {
+                            elements.push(format!("'{key}', json_build_object("))
+                        }
+
+                        QueryElement::ObjectClosingBoundary => {
+                            elements.push(")".to_string());
+
+                            if let Some(next_element) = peekable_elements.peek() {
+                                match next_element {
+                                    QueryElement::Field { .. }
+                                    | QueryElement::ObjectOpeningBoundary { .. } => {
+                                        elements.push(", ".to_string());
+                                    }
+                                    _ => {}
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        elements
+    }
+
+    fn get_topologically_sorted_joins(&mut self) -> Vec<JoinCondition> {
```

Comment:
Could use a helpful high-level rust doc comment here as well. The more info the better